### ISSUE TITLE
Feature suggestion: Add Common Platform Enumeration (CPE)

### ISSUE THREAD

**Issue by akunt** [AliasIO/Wappalyzer#1106]
The [CPE](https://cpe.mitre.org/about/) is a structured naming scheme for applications, see the [specification](https://cpe.mitre.org/specification/). 

For the lazy ones, version 2.3 uses at least the following attributes of an application, plenty more attributes are optional: 

`cpe:2.3:/<part>:<vendor>:<product>:<version>` 

Whereas `<part>` can be **a**pplication, **o**perating system or **h**ardware, so should always be **a** here. For most applications these CPEs already exist and can be found easily using the [search tool](https://web.nvd.nist.gov/view/cpe/search). For example for nginx version 1.5.4 the CPE would be:

`cpe:2.3:a:nginx:nginx:1.5.4:*:*:*:*:*:*:*` ([See](https://web.nvd.nist.gov/view/cpe/detail?keyword=nginx+1.5.4&status=FINAL&orderBy=CPEURI&namingFormat=2.3&cpeId=274258))

Adding this as optional information in the _apps.json_ would increase the API usability due to the standardized output. The `<version>` could be set dynamically. 

tl;dr: Add CPEs -> Profit

What do you guys think?

**Comment by faheel** [comment]
Seems like a good idea. Having the CPE data will be very useful for quite a few use cases. I would like to start working on this by adding CPEs for some of the most commonly used software (like MySQL, PHP, Django, Python, Linux, Windows etc.)

**Comment by AliasIO** [comment]
Hi @faheel. Sure, happy to include it.

**Comment by dhavalveera** [comment]
Hello

please can anyone let me know how to use CPE for Wappalyzer to Identify the Technology?


I am using Python - Flask & Django, and I want to know how & where to define CPE?



please tell me.
